compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10513,444 +10510,350 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
             *returnCond = TR::InstOpCode::COND_MASK0;
          }
       }
-   else if (isConstLength)
+   else
       {
-      if (node->getOpCode().hasSymbolReference() && node->getSymbolReference())
+      TR::LabelSymbol * shortCircuitLabel = generateLabelSymbol(cg);
+      TR::Register * tempSource1Reg = cg->evaluate(source1Node);
+      TR::Register * tempSource2Reg = cg->evaluate(source2Node);
+      if(isFoldedIf && isIfxcmpBrCondContainEqual)
          {
-         baseSource1Ref = generateS390MemoryReference(cg, node, source1Node, 0, true); // forSS=true
-         baseSource2Ref = generateS390MemoryReference(cg, node, source2Node, 0, true); // forSS=true
+         generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CLR, node, tempSource1Reg, tempSource2Reg, TR::InstOpCode::COND_BE, compareTarget);
          }
       else
          {
-         // memcmp2 lacks a symbol reference
-         baseSource1Ref = generateS390MemoryReference(cg->evaluate(source1Node), 0, cg);
-         baseSource2Ref = generateS390MemoryReference(cg->evaluate(source2Node), 0, cg);
-         cg->decReferenceCount(source1Node);
-         cg->decReferenceCount(source2Node);
+         generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CLR, node, tempSource1Reg, tempSource2Reg, TR::InstOpCode::COND_BE, shortCircuitLabel);
```

Comment:
Given that the only `targetLabel` is different in this `if-else` can we use the ternary?